### PR TITLE
[MeshingApplication] Adding computation of the element size in the mmg-level-set metric process

### PR DESCRIPTION
**Pull request by marcnunezc**
Dear all,

In this PR, I add a function in the level-set metric process to allow the user to define a MaxElementalSize, which results in a smoother remeshing around the zero-level. Before this PR, the element size used in the metric is a constant (inside the defined bound-layer) one that is given by the "minimal_size" parameter.

Here are two pictures in 2D with an embedded ellipse to visualize the difference:
Before this PR (BoundLayer=1.5, MinSize=0.01, the size inside the boundlayer is constant and equal to 0.01)
before_element_size

After this PR (BoundLayer=4.0, MinSize=0.001, MaxSize=1.0, the size inside the boundlayer varies from the maxsize to the minsize)
after_element_size

I added new parameters and member variables to allow for independent definitions of bound-layers, interpolation, etc for the element size and the anisotropic ratio. I also reduced the inputs in the anisotropic ratio function, those which were already member variables defined in the .h.

Variables and parameters names are just my initial choice, and I am completely open to change them if they are not clear enough or redundant.

### CODE REVIEW

File: applications/MeshingApplication/custom_processes/metrics_levelset_process.cpp [hunk 1]
```diff
@@ -29,6 +29,13 @@ ComputeLevelSetSolMetricProcess<TDim>::ComputeLevelSetSolMetricProcess(
     Parameters default_parameters = Parameters(R"(
     {
         "minimal_size"                         : 0.1,
+        "maximal_size"                         : 1.0,
+        "sizing_parameters":
+        {
+            "reference_variable_name"          : "DISTANCE",
+            "boundary_layer_max_distance"      : 1.0,
+            "interpolation"                    : "Constant"
```
Comment (philbucher) [inline applications/MeshingApplication/custom_processes/metrics_levelset_process.cpp]: please follow the style-guide:
```suggestion
            "interpolation"                    : "constant"
```

**Review by philbucher — COMMENTED** [review]
cool example!

**Review by loumalouomega — DISMISSED** [review]
I am OK, but maybe is a good idea to clean up so many member variables and use instead a global member variable Parameters

**Review by loumalouomega — DISMISSED** [review]
OK

**Comment by marcnunezc** [comment]
I forgot to update the mmg_process.py script!

**Review by loumalouomega — APPROVED** [review]
OK

**Comment by loumalouomega** [comment]
> I forgot to update the mmg_process.py script!

**Pull request merged**